### ISSUE TITLE
Problem with missing lang entries in Configuration

### ISSUE THREAD

**Issue by Matt75** [PrestaShop/PrestaShop#12962]
Hi,

I see on Core Weekly a problem in case of multishop with this https://github.com/PrestaShop/PrestaShop/pull/12666

I tested with a **multishop** and **multilang** instance
```sql
SELECT `id_lang`, `iso_code` FROM `ps_lang`
```
id_lang | iso_code
------------ | -------------
1 | fr
2 | de
```sql
SELECT `id_shop`, `id_shop_group`, `name` FROM `ps_shop`
```
id_shop | id_shop_group | name
------------ | ------------- | -------------
1 | 1 | Shop1
2 | 1 | Shop2
```sql
SELECT `id_lang`, `id_shop` FROM `ps_lang_shop`
```
id_lang | id_shop
------------ | -------------
1 | 1
1 | 2
2 | 2
```sql
SELECT `id_configuration`, `id_shop_group`, `id_shop`, `name`, `value`
FROM `ps_configuration`
WHERE `name` = 'PS_LANG_DEFAULT'
```
id_configuration | id_shop_group | id_shop | name | value
------------ | ------------- | ------------- | ------------- | -------------
1 | null | null | PS_LANG_DEFAULT | 1
380 | 1 | 2 | PS_LANG_DEFAULT | 2
```sql
SELECT `id_configuration`, `id_shop_group`, `id_shop`, `name`, `value`
FROM `ps_configuration`
WHERE `name` = 'PS_DELIVERY_PREFIX'
```
id_configuration | id_shop_group | id_shop | name | value
------------ | ------------- | ------------- | ------------- | -------------
42 | null | null | PS_DELIVERY_PREFIX | #FR
378 | 1 | 2 | PS_DELIVERY_PREFIX | null
```sql
SELECT `id_configuration`, `id_lang`, `value`
FROM `PS_configuration_lang`
WHERE `id_configuration` IN (SELECT `id_configuration` FROM `ps_configuration` WHERE `name` = 'PS_DELIVERY_PREFIX')
```
id_configuration | id_lang | value
------------ | ------------- | -------------
378 | 1 | #FR
378 | 2 | #DE
**So here there are missing entries for id_configuration equal to 42**

This query added in https://github.com/PrestaShop/PrestaShop/pull/12666 will fail
```sql
INSERT INTO `PREFIX_configuration_lang` (`id_configuration`, `id_lang`, `value`) VALUES
(
    SELECT
        `id_configuration`,
        (SELECT `value` FROM `PREFIX_configuration` WHERE `name` = 'PS_LANG_DEFAULT'),
        `value`
    FROM `PREFIX_configuration` c WHERE `name` = 'PS_DELIVERY_PREFIX'
    AND NOT EXISTS (SELECT 1 FROM `PREFIX_configuration_lang` WHERE `id_configuration` = c.`id_configuration`)
);
```
Because the subquery return more than one result

We must find another way to fix problem caused by missing entries in ps_configuration_lang for this configuration keys

Configuration key | Used in
------------ | -------------
PS_INVOICE_PREFIX | src/Adapter/Invoice/InvoiceOptionsConfiguration.php
PS_INVOICE_LEGAL_FREE_TEXT | src/Adapter/Invoice/InvoiceOptionsConfiguration.php
PS_INVOICE_FREE_TEXT | src/Adapter/Invoice/InvoiceOptionsConfiguration.php
PS_DELIVERY_PREFIX | src/Adapter/Order/Delivery/SlipOptionsConfiguration.php
PS_RETURN_PREFIX | controllers/admin/AdminReturnController.php
PS_SEARCH_BLACKLIST | controllers/admin/AdminSearchConfController.php
PS_CUSTOMER_SERVICE_SIGNATURE | controllers/admin/AdminCustomerThreadsController.php
PS_MAINTENANCE_TEXT | src/Adapter/Shop/MaintenanceConfiguration.php
PS_LABEL_IN_STOCK_PRODUCTS | src/Adapter/Product/StockConfiguration.php
PS_LABEL_OOS_PRODUCTS_BOA | src/Adapter/Product/StockConfiguration.php
PS_LABEL_OOS_PRODUCTS_BOD | src/Adapter/Product/StockConfiguration.php

Because on migrated controllers it cause an error
```
Symfony\Component\Form\Exception\UnexpectedTypeException:
Expected argument of type "object, array or empty", "string" given
```

**Comment by rblaurin** [comment]
@Matt75, I made some changes to the SQL code. What do you think now?
```
INSERT INTO `ps_configuration_lang` (`id_configuration`, `id_lang`, `value`)
SELECT `id_configuration`, l.`id_lang`, `value` 
FROM `ps_configuration` c
JOIN `ps_lang_shop` l on l.`id_shop` = coalesce(c.`id_shop`, 1)
WHERE `name` = 'PS_DELIVERY_PREFIX' AND NOT EXISTS (SELECT 1 FROM `ps_configuration_lang` WHERE `id_configuration` = c.`id_configuration`)
```

**Comment by Matt75** [comment]
@rblaurin Yes this should work, I guess

But this fix doen't go to `1.7.0.0.sql` because issues related to this errors comes from people who already use PrestaShop 1.7 and these people have to execute the fix
So https://github.com/PrestaShop/PrestaShop/pull/12666 should be reverted and another should be create in `1.7.6.0.sql` to fix all this configuration keys :

Configuration key | Used in
------------ | -------------
PS_INVOICE_PREFIX | src/Adapter/Invoice/InvoiceOptionsConfiguration.php
PS_INVOICE_LEGAL_FREE_TEXT | src/Adapter/Invoice/InvoiceOptionsConfiguration.php
PS_INVOICE_FREE_TEXT | src/Adapter/Invoice/InvoiceOptionsConfiguration.php
PS_DELIVERY_PREFIX | src/Adapter/Order/Delivery/SlipOptionsConfiguration.php
PS_RETURN_PREFIX | controllers/admin/AdminReturnController.php
PS_SEARCH_BLACKLIST | controllers/admin/AdminSearchConfController.php
PS_CUSTOMER_SERVICE_SIGNATURE | controllers/admin/AdminCustomerThreadsController.php
PS_MAINTENANCE_TEXT | src/Adapter/Shop/MaintenanceConfiguration.php
PS_LABEL_IN_STOCK_PRODUCTS | src/Adapter/Product/StockConfiguration.php
PS_LABEL_OOS_PRODUCTS_BOA | src/Adapter/Product/StockConfiguration.php
PS_LABEL_OOS_PRODUCTS_BOD | src/Adapter/Product/StockConfiguration.php

@Quetzacoalt91 WDYT ?

**Comment by Quetzacoalt91** [comment]
This makes sense and we can move forward.

However, regarding the deadlines about 1.7.6.0, I guess this is too late for having this improvement in it. We can revert this PR and take the time we need to make another one for 1.7.7.0.

**Comment by rblaurin** [comment]
> But this fix doen't go to `1.7.0.0.sql` because issues related to this errors comes from people who already use PrestaShop 1.7 and these people have to execute the fix

Yes, that's right. I will adapt the code for all those configuration keys and make another PR then.
